General.ruleset

```diff
@@ -0,0 +1,110 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
```

Comment:
I get such warnings from somewhere:

![image](https://user-images.githubusercontent.com/1976647/88099361-5be00980-cb9b-11ea-8a0b-85267d2d233f.png)

Reply:
In what case do you get this warning? When you open ruleset within Visual Studio with solution that has projects referencing used analyzers, everything should be OK. It required analyzer packages to be somewhere.

Reply:
I tested it with the Finitive solution in this branch: https://github.com/Lombiq/Finitive/tree/issue/FINI-145 Note that for this to work I've already changed these (but didn't commit):
![image](https://user-images.githubusercontent.com/1976647/88204081-18dd6f00-cc4b-11ea-9a19-e654bf4be490.png)

I get this when I open the solution and it never goes away.

Reply:
I've pushed this so I can work on custom analyzers in another branch.

Reply:
I think I get it:

- That I didn't always get results for a whole solution is simply because after a rebuild it takes minutes to generate the warnings. Note that I need to do a rebuild, just building (F6) an already built solution doesn't bring up all the warnings from the solution. So that's clear, nothing to do, though it's strange it needs a rebuild (and if I close and reopen the solution where I've done a rebuild and seen all the warnings previously, I need to rebuild again).
- I get the "An instance of analyzer..." warnings when opening the solution after a `git clean -fxd`. It goes away after fully building the solution, then reopening it. This is correlated with the `System.TypeLoadException` I mentioned above. However, then eventually (perhaps after reopening the solution a few times) I get the same warnings for the same (already fully built, previously full working) solution. And then I do a rebuild, get all the warnings but also get the `System.TypeLoadException` ones and thus IntelliSense suggestions are broken... So this is flaky.

Reply:
I don't know what's happening here. Opening the solution today doesn't show the `System.TypeLoadException` warnings, even though I didn't do anything between now and the warnings being there, apart from shutting down and starting my PC. Don't you experience the same? Like when you freshly clone the Finitive repo, checkout issue/FINI-145, and open the solution?

Reply:
I don't remember either I had `System.TypeLoadException`.

_Like when you freshly clone the Finitive repo, checkout issue/FINI-145, and open the solution?_
Did it, built solution and everything is fine.

Reply:
Hmm, let me ask my colleagues, it may be due to the accented character in the path ("Zoltán") though it going away after a while suggests it's not because of that.

Reply:
Are you on VS 16.6.5 Pro too?

Reply:
It turns out this was the Microsoft Code Analysis extension. I've added a note on this into the Readme.